File: BusInfo/app/src/main/java/com/daeseong/businfo/SplashActivity.java
package com.daeseong.businfo;

import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Handler;
import android.os.Looper;
import android.util.Log;
import android.view.Window;
import android.view.WindowManager;
import java.lang.ref.WeakReference;

public class SplashActivity extends AppCompatActivity {

    private static final String TAG = SplashActivity.class.getSimpleName();

    private Handler handler = new Handler(Looper.getMainLooper());

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        InitTitleBar();

        setContentView(R.layout.activity_splash);

        init();
    }

    private void init(){

        WeakReference<SplashActivity> activityReference = new WeakReference<>(this);

        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                SplashActivity activity = activityReference.get();
                if (activity != null) {
                    activity.startActivity(new Intent(activity, MainActivity.class));
                    activity.finish();
                }
            }
        }, 1000);
    }

    private void InitTitleBar() {

        try {
            //안드로이드 8.0 오레오 버전에서만 오류 발생
            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
            supportRequestWindowFeature(Window.FEATURE_NO_TITLE);
        } catch (Exception ex) {
        }

        try {
            requestWindowFeature(Window.FEATURE_NO_TITLE);
            getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
        } catch (Exception ex) {
        }
    }

    @Override
    public void onBackPressed() {
        //super.onBackPressed();
        return;
    }

    @Override
    protected void onPause() {
        super.onPause();

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        try {
            if (handler != null) {
                handler.removeCallbacksAndMessages(null);
            }
        } catch (Exception ex) {
            Log.e(TAG, ex.getMessage());
        }
    }
}

File: BusInfo(kotlin)/app/src/main/java/com/daeseong/businfo/SplashActivity.kt
package com.daeseong.businfo

import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.Window
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import java.lang.ref.WeakReference

class SplashActivity : AppCompatActivity() {

    private val tag = SplashActivity::class.java.simpleName

    private val handler = Handler(Looper.getMainLooper())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initTitleBar()

        setContentView(R.layout.activity_splash)

        init()
    }

    private fun init() {
        val activityReference = WeakReference(this)

        handler.postDelayed({
            val activity = activityReference.get()
            activity?.run {
                startActivity(Intent(this, MainActivity::class.java))
                finish()
            }
        }, 1000)
    }

    private fun initTitleBar() {
        try {
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
            supportRequestWindowFeature(Window.FEATURE_NO_TITLE)
        } catch (ex: Exception) {
        }

        try {
            requestWindowFeature(Window.FEATURE_NO_TITLE)
            window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
        } catch (ex: Exception) {
        }
    }

    override fun onBackPressed() {
        //super.onBackPressed()
        return
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onDestroy() {
        super.onDestroy()

        try {
            handler.removeCallbacksAndMessages(null)
        } catch (ex: Exception) {
        }
    }
}
